hnn_core/network.py

```diff
@@ -353,6 +355,12 @@ def __init__(self, params, add_drives_from_params=False, legacy_mode=True):
         # XXX this can be removed once tests are made independent of HNN GUI
         # creates nc_dict-entries for ALL cell types
         self._legacy_mode = legacy_mode
+        if self._legacy_mode:
+            warnings.warn(
+                'Legacy mode is used solely to maintain compatibility with'
```

Comment:
@jasmainak what do you think of this? I'm ok with being annoying if you are :wink:

Reply:
okay for me ! what happens during optimization though? Hope the warning is emitted only once?

Reply:
That's a good point - I'm assuming this will get printed anytime `Network.copy()` is called?

Reply:
Alternatively, we could place this warning in `jones_2009_model()` since all the other template models call that function

Reply:
My reasoning was they all have legacy mode as well as an option